content/chapters/io/lecture/demo/devices/hwaddr-ioctl.c

```diff
@@ -16,31 +16,36 @@
 
 #include "utils/utils.h"
 
-#define IFNAME			"enp0s31f6"
-
-int main(void)
+int main(int argc, const char *argv[])
 {
+	if (argc < 2) {
+		printf("Usage: %s interface_name\n", argv[0]);
+		return 0;
+	}
 	int s;
 	struct ifreq ifr;
 	int rc;
 	size_t i;
+	char *ifname = malloc(strlen(argv[1]));
+	strcpy(ifname, argv[1]);
```

Comment:
No need to copy `argv[1]` to the heap. Just use it as-is below.

Reply:
Done. I will remember this for future PRs.